django/db/migrations/serializer.py

```diff
@@ -405,6 +427,10 @@ def serializer_factory(value):
     # Anything that knows how to deconstruct itself.
     if hasattr(value, "deconstruct"):
         return DeconstructibleSerializer(value)
+    if hasattr(types, "GenericAlias") and isinstance(value, types.GenericAlias):
+        return GenericAliasSerializer(value)
+    elif hasattr(value, "__origin__") and hasattr(value, "__args__"):
+        return GenericAliasSerializer(value)
```

Comment:
You can just add `types.GenericAlias: GenericAliasSerializer` to the serializer registry, above `collections.abc.Iterable: IterableSerializer,` to ensure it's matched first.